### PR TITLE
Bug Fixes:

### PR DESCRIPTION
**Pull request by cs-ankit-gupta**
Descriptions:
1. 0931187 - OT_asset_management: Task marked completed from asset change activity should be listed in the collaboration tab of asset
2. 0931189 - OT asset management: Asset change activity comments needs to be reformatted.
3. 0931205 - OT asset management: Need to add the comment in collaboration tab on cancellation of add new task from template.
4. 0931209 - OT asset management: Scenario - OT - Asset Change Activity PB  changes.
5. 0931708 - OT_asset_managment: Need to provide an appropriate error message if user runs Alert scenario prior to asset scenario.

Fix:
1. Added comment on assets on completing task under playbook "Add Cyber Asset", "High Impact Baseline Change", "Medium Impact Baseline Change", "Remove Cyber Asset", "Replace Cyber Asset", and "Modify Cyber Asset"
2. Reformatted task comment on Asset Change Activity record.
3. Added comment if "Add Task Template" got cancelled.
4. Changed step name from "Add New Asset" to "Add new Cyber Asset" and updated uniqueness constraint.
5. Added Logic to check if OT Sample if simulated or not. If not then create sample assets related to Alert in OT - Add Sample Alerts

### CODE REVIEW

File: playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json [hunk 1]
```diff
@@ -15,6 +15,102 @@
     "versions": [],
     "triggerStep": "\/api\/3\/workflow_steps\/f6e602b6-109f-4bd8-bf3f-28ce08117592",
     "steps": [
+        {
+            "@type": "WorkflowStep",
+            "name": "Get All Incomplete Task",
```
Comment (cs-suyog-jinturkar) [inline playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json]: Task --> Tasks

File: playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json [hunk 2]
```diff
@@ -15,6 +15,102 @@
...
+            "status": null,
+            "top": "165",
+            "left": "475",
+            "stepType": "\/api\/3\/workflow_step_types\/b593663d-7d13-40ce-a3a3-96dece928770",
+            "group": null,
+            "uuid": "6f263d5c-06e0-47ff-8a3e-99867c8ad347"
+        },
+        {
+            "@type": "WorkflowStep",
+            "name": "Incomplete Task Comment",
```
Comment (cs-suyog-jinturkar) [inline playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json:55]: imo, Desc and name is not matching

File: playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json [hunk 3]
```diff
@@ -97,13 +193,43 @@
...
+            "targetStep": "\/api\/3\/workflow_steps\/110145ac-6461-473f-b3c5-d82ab529d4ce",
+            "sourceStep": "\/api\/3\/workflow_steps\/6f263d5c-06e0-47ff-8a3e-99867c8ad347",
+            "label": null,
+            "isExecuted": false,
+            "group": null,
+            "uuid": "befccc10-e4fc-4725-8950-d6719061559c"
+        },
+        {
+            "@type": "WorkflowRoute",
+            "name": "Is Incomplete Task Exists -> Copy of Update Asset Change Activity",
```
Comment (cs-suyog-jinturkar) [inline playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json]: Copy of Update Asset Change Activity -->> ??

File: playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json [hunk 4]
```diff
@@ -97,13 +193,43 @@
...
             "targetStep": "\/api\/3\/workflow_steps\/5c7ff996-4f2e-4865-906f-173b691fc4eb",
+            "sourceStep": "\/api\/3\/workflow_steps\/110145ac-6461-473f-b3c5-d82ab529d4ce",
+            "label": "No",
+            "isExecuted": false,
+            "group": null,
+            "uuid": "707bf21d-41a6-4216-b84d-e9d7ec6a35b7"
+        },
+        {
+            "@type": "WorkflowRoute",
+            "name": "Start -> Get All Open Task",
```
Comment (cs-suyog-jinturkar) [inline playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json]: Task --> Tasks

File: playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json (hunk 2 above)
Comment (cs-ankit-gupta) [reply playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json:55]: incorporated

File: playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json (hunk 1 above)
Comment (cs-ankit-gupta) [reply playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json]: incorporated

File: playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json (hunk 3 above)
Comment (cs-ankit-gupta) [reply playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json]: incorporated

File: playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json (hunk 4 above)
Comment (cs-ankit-gupta) [reply playbooks/02 - Use Case - Asset Change Activity/Manage Asset Change Activity Closure.json]: incorporated

File: playbooks/02 - Use Case - Asset Change Activity/Add Cyber Asset.json [hunk 5]
```diff
@@ -5289,27 +5291,238 @@
...
-            "left": "125",
+            "top": "1110",
+            "left": "475",
             "stepType": "\/api\/3\/workflow_step_types\/dc6ac63d-c5a5-472f-9eb4-6b18473a98b8",
             "group": null,
             "uuid": "890ce319-249e-477b-9523-1de9ca1c30f7"
         },
+        {
+            "@type": "WorkflowStep",
+            "name": "Comment Approver Approval To Implement Task Complete",
```
Comment (cs-suyog-jinturkar) [inline playbooks/02 - Use Case - Asset Change Activity/Add Cyber Asset.json]: Didn't get this playbook name. Why description is empty?

File: playbooks/02 - Use Case - Asset Change Activity/Add Cyber Asset.json [hunk 6]
```diff
@@ -5289,27 +5291,238 @@
...
+            "status": null,
+            "top": "1660",
+            "left": "820",
+            "stepType": "\/api\/3\/workflow_step_types\/2597053c-e718-44b4-8394-4d40fe26d357",
+            "group": null,
+            "uuid": "2273962c-205a-4f44-98d3-70a49644a5ab"
+        },
+        {
+            "@type": "WorkflowStep",
+            "name": "Comment Assignee Approval To Implement Task Complete",
```
Comment (cs-suyog-jinturkar) [inline playbooks/02 - Use Case - Asset Change Activity/Add Cyber Asset.json]: same as above?

File: playbooks/02 - Use Case - Asset Change Activity/High Impact Baseline Change.json [hunk 7]
```diff
@@ -7557,13 +7916,13 @@
         },
         {
             "@type": "WorkflowRoute",
-            "name": "Copy of Assignee Approval To Implement -> Copy 1 of Is Approver Assigned",
-            "targetStep": "\/api\/3\/workflow_steps\/63692e13-58e2-45b1-a729-f571fd1d5f3e",
-            "sourceStep": "\/api\/3\/workflow_steps\/3558e78e-ba4c-4957-8b46-9f9c8c4e828d",
+            "name": "Copy of Comment Approver Approval To Test Task Complete -> Assignee Approval To Implement",
```
Comment (cs-suyog-jinturkar) [inline playbooks/02 - Use Case - Asset Change Activity/High Impact Baseline Change.json]: copy?

File: playbooks/02 - Use Case - Asset Change Activity/Medium Impact Baseline Change.json [hunk 8]
```diff
@@ -5677,13 +5889,13 @@
         },
         {
             "@type": "WorkflowRoute",
-            "name": "Approver Approval To Implement -> Implement Change",
-            "targetStep": "\/api\/3\/workflow_steps\/683afbc0-4503-4bcf-ba8b-0c925bc9ff6d",
+            "name": "Approver Approval To Implement -> Copy of Comment Assignee Approval To Implement Task Complete",
```
Comment (cs-suyog-jinturkar) [inline playbooks/02 - Use Case - Asset Change Activity/Medium Impact Baseline Change.json]: copy??

File: playbooks/02 - Use Case - Asset Change Activity/Medium Impact Baseline Change.json [hunk 9]
```diff
@@ -5697,93 +5909,113 @@
...
-            "uuid": "62d43b6c-88e9-4925-9aae-166401af2cc0"
+            "uuid": "2c7a071e-6a23-41bb-940e-c78ff0f90bcc"
         },
         {
             "@type": "WorkflowRoute",
-            "name": "Assignment And Approver Name -> Do Nothing",
-            "targetStep": "\/api\/3\/workflow_steps\/f6fc3f3d-fa02-4943-9c66-0e50cc903a75",
-            "sourceStep": "\/api\/3\/workflow_steps\/d2aa1894-7f18-4ec6-980d-2bfe0ebd64bf",
-            "label": "Cancel",
+            "name": "Copy of Check Approver 2 -> Implement Change",
```
Comment (cs-suyog-jinturkar) [inline playbooks/02 - Use Case - Asset Change Activity/Medium Impact Baseline Change.json]: copy?

File: playbooks/02 - Use Case - Asset Change Activity/Add Cyber Asset.json (hunk 5 above)
Comment (cs-ankit-gupta) [reply playbooks/02 - Use Case - Asset Change Activity/Add Cyber Asset.json]: Incorporated

File: playbooks/02 - Use Case - Asset Change Activity/Add Cyber Asset.json (hunk 6 above)
Comment (cs-ankit-gupta) [reply playbooks/02 - Use Case - Asset Change Activity/Add Cyber Asset.json]: Incorporated

File: playbooks/02 - Use Case - Asset Change Activity/High Impact Baseline Change.json (hunk 7 above)
Comment (cs-ankit-gupta) [reply playbooks/02 - Use Case - Asset Change Activity/High Impact Baseline Change.json]: Incorporated

File: playbooks/02 - Use Case - Asset Change Activity/Medium Impact Baseline Change.json (hunk 8 above)
Comment (cs-ankit-gupta) [reply playbooks/02 - Use Case - Asset Change Activity/Medium Impact Baseline Change.json]: Incorporated

File: playbooks/02 - Use Case - Asset Change Activity/Medium Impact Baseline Change.json (hunk 9 above)
Comment (cs-ankit-gupta) [reply playbooks/02 - Use Case - Asset Change Activity/Medium Impact Baseline Change.json]: by default it gives copy if any step is copied, BTW - Incorporated

**Review by cs-suyog-jinturkar — APPROVED** [review]

**Pull request merged**